### ISSUE TITLE
'Is there a missing '+' on the previous line?' on "use script"

### ISSUE THREAD

**Issue by kofifus** [google/closure-compiler#2046]
I combine a few js files together and end up "use strict" after code, for example:

```
function bla() {}
"use strict";
...
```

closure-compiler then complains:

```
test.js:2: WARNING - Suspicious code. Is there a missing '+' on the previous line?
"use strict";
^

```

How can I remove this warning ?

**Comment by MatrixFrog** [comment]
That shouldn't happen. Is it possible you're using an old version of the compiler?

**Comment by kofifus** [comment]
@java -jar compiler.jar --version
Closure Compiler (http://github.com/google/closure-compiler)
Version: v20160911
Built on: 2016-09-13 16:51

(on windows)

**Comment by MatrixFrog** [comment]
Strange. Can you post the full file that you're passing to the compiler (or a smaller one that still reproduces the issue)?

**Comment by kofifus** [comment]
smaller than that ? it's two lines ...

**Comment by MatrixFrog** [comment]
I just noticed I misread the original example. This is a correct warning because 'use strict' doesn't make sense as a directive unless it's the first line of a file or the first line of a function.

**Comment by kofifus** [comment]
so how do I deal with a situation of combining multiple js files each with "use strict" at the top and then sending them to closure-compiler ?

**Comment by Dominator008** [comment]
@kofifus Sounds like you will benefit from some kind of module system, but you can do `--jscomp_off=suspiciousCode` for now.

**Comment by MatrixFrog** [comment]
Or pass the files separately rather than cating them together first, perhaps.

**Comment by ChadKillingsworth** [comment]
Or wrap them in IIFE's:

```js
(function() {

}).call(this)
```

**Comment by kofifus** [comment]
yes there are way around it, but what I'm doing is valid javascript and it's a real hassle to change my build system just to remove this warning which in my case does not make sense...

**Comment by Dominator008** [comment]
The warning message comes from here:
https://github.com/google/closure-compiler/blob/1163af7a0dd83ce4b9f4e3952a3349fb41bdcfc8/src/com/google/javascript/jscomp/CheckSideEffects.java#L123
It means that the compiler detected a "standalone" string literal that is not used anywhere. In many cases, it could mean that the programmer missed a `+` on the previous line, like in:
```js
var foo = 'This '
'is a string.';
```
While the warning message is not the most intuitive in your case, it is generally helpful.
IIRC there's currently no way to suppress it "just for this case". You can put an `@suppress {uselessCode}` on the top of that file but it will also suppress other warnings for useless code.

**Comment by kofifus** [comment]
thx Dominator008, so can I suggest this msg to be supressed for if the string is "use script"; ?

**Comment by Dominator008** [comment]
@kofifus It is still a standalone string literal that is not used anywhere. The warning is correct.

**Comment by kofifus** [comment]
what I'm doing is very reasonable and common place, concat a bunch of js and run the compiler on them surely you can see that such warnings for "use script"; should be ignored ?
